app/assets/stylesheets/components/button.scss

```diff
@@ -0,0 +1,41 @@
+.button {
```

Comment:
1. what does `$self: &;` do in this class, that is not already done by `&`
2. are these rules arranged by Position / Box Model / Typography / Visual / Misc

Reply:
1. it does nothing at the moment; I should remove it. 
2. They should be, although I do fudge the order on occasion. Is there line that looks off?
4. According to [this](https://blog.codinghorror.com/of-spaces-underscores-and-dashes/), dashes are the proper separator for filenames.

Reply:
Approved.